### PR TITLE
PCHR-1861: Handle exception on contract create and update

### PR DESCRIPTION
**Pull request by bmnepali**
### Issue:
The exception while creating is captured but not handled properly in ` hrjobcontract/js/src/job-contract/controllers/modal/modal-contract-new.js` and  `hrjobcontract/js/src/job-contract/controllers/modal/modal-contract.js` in `save()` and `contractEdit()` functions respectively.

### Solution:
1. Add code to display the exception properly in `hrjobcontract/js/src/job-contract/controllers/modal/modal-contract-new.js` in save() functions as below:
```
  function (reason) {
      CRM.alert(reason, 'Error', 'error');
      ContractService.delete(contractId);
      $scope.$broadcast('hrjc-loader-hide');
  } 
```
1.1 This will also delete the contract just created if the exception happens in the middle of saving contract data.

1.2 Hide the spinner, allowing user to edit the changes in the contract modal.

2. Remove `$modalInstance.dismiss();` from error cache block, which dismisses the modal in contractEdit() function form `hrjobcontract/js/src/job-contract/controllers/modal/modal-contract.js` in `contractEdit()` function, allowing user to edit the changes in the contract modal.

### Screenshots: (gif)
1. Displays error appropriately on creating contract:
add contract-iloveimg-compressed

2. Displays error appropriately on saving  contract after editing it:
edit contract-iloveimg-compressed

3. Displays error if contract cannot be deleted when an exception occurs.
display-contract-delete-error

4. Reformat Exception message to display in more readable format.
screen shot 2017-04-11 at 9 42 04 pm

### CODE REVIEW

File: hrjobcontract/js/src/job-contract/controllers/modal/modal-contract-new.js [hunk 1]
```diff
@@ -217,18 +217,17 @@ define([
...
-                return $q.reject();
               }).then(function() {
                 $scope.$broadcast('hrjc-loader-hide');
                 $modalInstance.close(contract);
 
                 pubSub.publish('contract:created', settings.contactId);
                 pubSub.publish('contract-refresh');
+              }, function (reason) {
+                CRM.alert(reason, 'Error', 'error');
+                ContractService.delete(contractId);
```
Comment (dhavalsharma) [inline hrjobcontract/js/src/job-contract/controllers/modal/modal-contract-new.js]: is delete sync call?

File: hrjobcontract/js/src/job-contract/controllers/modal/modal-contract.js [hunk 2]
```diff
@@ -366,7 +366,6 @@ define([
         }, function(reason) {
           $scope.$broadcast('hrjc-loader-hide');
           CRM.alert(reason, 'Error', 'error');
-          $modalInstance.dismiss();
         });
```
Comment (dhavalsharma) [inline hrjobcontract/js/src/job-contract/controllers/modal/modal-contract.js]: modal is no more dismissed?

Comment (bmnepali) [inline hrjobcontract/js/src/job-contract/controllers/modal/modal-contract.js]: Yeah! we want it to be displayed, so the user can make modification to data, when the exception is  captured.

File: hrjobcontract/js/src/job-contract/controllers/modal/modal-contract-new.js [hunk 3]
```diff
@@ -226,8 +226,12 @@ define([
               },
               function (reason) {
                 CRM.alert(reason, 'Error', 'error');
-                ContractService.delete(contractId);
-                $scope.$broadcast('hrjc-loader-hide');
+                ContractService.delete(contractId).then(function(result){
```
Comment (dhavalsharma) [inline hrjobcontract/js/src/job-contract/controllers/modal/modal-contract-new.js]: please add error handler also
`function(result){}, function(error){}`

Comment (bmnepali) [inline hrjobcontract/js/src/job-contract/controllers/modal/modal-contract-new.js]: Added!

File: hrjobcontract/js/src/job-contract/services/utils.js [hunk 4]
```diff
@@ -152,16 +152,23 @@ define([
 
       },
       errorHandler: function(data, msg, deffered) {
+        debugger
```
Comment (dhavalsharma) [inline hrjobcontract/js/src/job-contract/services/utils.js]: ps remove

Comment (bmnepali) [inline hrjobcontract/js/src/job-contract/services/utils.js]: done!

**Pull request merged**